source/functions/test.txt

```diff
@@ -0,0 +1,313 @@
+.. _test-functions:
+
+====================
+Test Atlas Functions
+====================
+
+.. contents:: On this page
+   :local:
+   :backlinks: none
+   :depth: 2
+   :class: singlecol
+
+This page describes some strategies you can use to test your Atlas Functions.
+
+Due to some differences between the the Functions
+JavaScript runtime and the standard Node.js runtime, you must take some unique considerations
+into account when testing Functions. This page covers how to handle
+the uniqueness of Functions.
+
+Before You Begin
+----------------
+
+#. :ref:`Create an App Services App <create-a-realm-app>`
+#. :ref:`Configure the CLI to work with your App <realm-cli>` or
+   :ref:`automate deployment with Github <deploy-github>`
+
+.. _unit-test-functions:
+
+Unit Tests for Functions
+------------------------
+
+You can write unit tests to validate the functionality of your Functions.
+Use any Node.js-compatible testing framework to test Functions.
+The examples on this page use the `Jest testing framework <https://jestjs.io/>`__.
+
+You must use :wikipedia:`CommonJS modules <CommonJS>` to write unit tests for
+Functions.
+
+.. procedure::
+
+   .. step:: Get a local copy of your App Services App
+
+      .. tabs::
+
+         .. tab:: {+cli+}
+            :tabid: cli
+
+            Pull the latest configuration of your App from the server.
+
+            .. code-block:: sh
+
+              realm-cli pull --remote <App ID>
+
+         .. tab:: Github
+            :tabid: github
+
+            Pull the latest configuration of your App from Github.
+
+            .. code-block:: sh
+
+              git pull <Remote Name> <Branch name>
+
+
+   .. step:: Create a new Function
+
+      Create a new Function. In the App's configuration files, create a new JavaScript
+      file in the ``functions`` directory for your function.
+
+      .. code-block:: sh
+
+         touch functions/hello.js
+
+
+      You also need to add configuration information for the Function to ``function/config.json``.
+
+      .. code-block:: json
+
+          {
+              "name": "hello",
+              "private": false,
+              "run_as_system": true
+          },
+
+      .. seealso::
+
+         For more information on creating a new Function,
+         refer to :ref:`Define a Function <define-a-function>`.
+
+   .. step:: Write Function code
+
+      To make your Function code easy to test, keep it modular
+      by separating its concerns into distinct components.
+      You must keep all logic for the Function in the file you defined
+      in the previous step. You cannot perform relative imports from other files
+      in your project in a Function file.
+      You can also :ref:`import dependencies using npm <external-dependencies>`.
+
+      You must export your function by assigning it to ``export``.
```

Comment:
```suggestion
      You must export your main function by assigning it to ``export``.
```
not sure if this would be more clear - trying to disambiguate between JS functions and AS Functions a bit beyond just capitalization.